File: FitBook.Model/Responses/Trainings/TrainingResponse.cs
namespace FitBook.Model.Responses.Trainings;

public class TrainingResponse : IEntityResponse
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public int DurationMinutes { get; set; }
    public int MaxParticipants { get; set; }
    public bool IsActive { get; set; }
    public int TrainingCategoryId { get; set; }
    public string TrainingCategoryName { get; set; } = string.Empty;
    public int DifficultyLevelId { get; set; }
    public string DifficultyLevelName { get; set; } = string.Empty;
    public DateTime CreatedAtUtc { get; set; }
    public DateTime? UpdatedAtUtc { get; set; }
}

File: UI/fitbook_desktop/lib/models/responses/training_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'training_response.g.dart';

/// Mirrors `FitBook.Model.Responses.TrainingResponse`.
@JsonSerializable()
class TrainingResponse {
  TrainingResponse({
    required this.id,
    required this.name,
    required this.description,
    required this.durationMinutes,
    required this.maxParticipants,
    required this.isActive,
    required this.trainingCategoryId,
    required this.trainingCategoryName,
    required this.difficultyLevelId,
    required this.difficultyLevelName,
    required this.createdAtUtc,
    this.updatedAtUtc,
  });

  final int id;
  final String name;
  final String description;
  final int durationMinutes;
  final int maxParticipants;
  final bool isActive;
  final int trainingCategoryId;
  final String trainingCategoryName;
  final int difficultyLevelId;
  final String difficultyLevelName;
  final DateTime createdAtUtc;
  final DateTime? updatedAtUtc;

  factory TrainingResponse.fromJson(Map<String, dynamic> json) =>
      _$TrainingResponseFromJson(json);

  Map<String, dynamic> toJson() => _$TrainingResponseToJson(this);
}
